### PR TITLE
finding the ports by ID insted of an index

### PR DESCRIPTION
**Pull request by filipeotero**
### Purpose

It's included in this PR the fix for a bug at the outputs when a group is collapsed. 

fixOutPutCollapsedBar

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang 

### FYIs

@RobertGlobant20

### CODE REVIEW

**Comment by reddyashish** [comment]
How would it look when the subgroup is collapsed and then the parent group is collapsed?

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 1]
```diff
@@ -759,14 +759,18 @@ private List<PortViewModel> CreateProxyOutPorts(IEnumerable<PortModel> groupPort
...
+            int index = 0;
+            foreach (var group in groupPortModels)
             {
-                var model = groupPortModels.ElementAt(i);
-                newPortViewModels.Add(originalPortViewModels[i].CreateProxyPortViewModel(model));
-
-                // calculate new position for the proxy outports
-                model.Center = CalculatePortPosition(model, i);
-                model.Owner.ReportPosition();
+                var originalPort = originalPortViewModels.FirstOrDefault(x => x.PortModel.GUID == group.GUID);
```
Comment (reddyashish) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:765]: I am just curious, how does this make a difference. Weren't were iterating through all the output-ports in the first place? or is this just refactoring?

Comment (filipeotero) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:765]: Hi, thanks for the observation. Previously we were interacting through all the ports without controlling its owner. Some ports were wrongly set to another group or node, causing the bug. By searching its GUID we make sure the right node/group is referenced.

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 2]
```diff
@@ -771,13 +771,18 @@ private List<PortViewModel> CreateProxyOutPorts(IEnumerable<PortModel> groupPort
                 .ToList();
 
             var newPortViewModels = new List<PortViewModel>();
-            for (int i = 0; i < groupPortModels.Count(); i++)
+            int index = 0;
+            foreach (var group in groupPortModels)
```
Comment (zeusongit) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:775]: Any specific reason to switch to this foreach loop? we already had index variable in the previous loop.

**Comment by filipeotero** [comment]
> Looks good to me after the conflicts are resolved. One minor suggestion: If we can have little vertical margin between the CBN ports, I feel it would be better.

I also think it looks better. Here it is:

image

**Comment by QilongTang** [comment]
@reddyashish @zeusongit Is this PR approval?
Also @filipeotero Would you record a final version of the gif for me so I can add for the demo? We can use the nested group case, collapse subgroup first then parent group one

**Review by zeusongit — APPROVED** [review]

**Review by reddyashish — APPROVED** [review]

**Comment by QilongTang** [comment]
Thanks @zeusongit and @reddyashish for the review. Merging.

**Pull request merged**